packages/cozy-harvest-lib/src/models/ConnectionFlow.js

```diff
@@ -440,31 +440,51 @@ export class ConnectionFlow {
         )
       }
 
-      logger.debug('ConnectionFlow: Creating/updating account...', account)
-
-      account = await createOrUpdateAccount({
-        account,
-        cipher,
-        client,
-        flow: this,
-        konnector,
-        konnectorPolicy,
-        userCredentials
-      })
+      if (konnectorPolicy.needsAccountAndTriggerCreation) {
+        logger.debug('ConnectionFlow: Creating/updating account...', account)
+        account = await createOrUpdateAccount({
+          account,
+          cipher,
+          client,
+          flow: this,
+          konnector,
+          konnectorPolicy,
+          userCredentials
+        })
 
-      this.account = account
+        this.account = account
 
-      logger.info(`ConnectionFlow: Saved account ${account._id}`)
+        logger.info(`ConnectionFlow: Saved account ${account._id}`)
 
-      if (!konnectorPolicy.needsTriggerCreation) {
-        await this.ensureTriggerAndLaunch(client, {
+        logger.debug('ConnectionFlow: Ensuring trigger...')
+        const ensuredTrigger = await ensureTrigger(client, {
```

Comment:
why this change in this commit?

Reply:
Also, it should be called... `trigger` no?

Reply:
I think it is clearer to separate account creation, trigger creation and trigger launch. And this makes it possible to policies to activated/deactivate any of them.